.github/workflows/ci.yml

```diff
@@ -53,69 +99,64 @@ jobs:
         with:
           dotnet-version: 5.0.x
       - name: Download Consul
+        if: matrix.component == 'Consul'
         shell: bash
         run: |
           cd Consul.Test
           SYSTEM=$(uname | sed 's/MINGW.*/windows/' | tr A-Z a-z)
-          curl -sSL https://releases.hashicorp.com/consul/${CONSUL_VERSION}/consul_${CONSUL_VERSION}_${SYSTEM}_amd64.zip -o consul.zip
+          curl -sSL https://releases.hashicorp.com/consul/${{ matrix.consul }}/consul_${{ matrix.consul }}_${SYSTEM}_amd64.zip -o consul.zip
           unzip consul.zip
           rm consul.zip
-      - name: Build
-        run: dotnet build --configuration=Release -p:TreatWarningsAsErrors=true
+      - name: Build library
+        run: dotnet build --framework ${{ steps.dotnet.outputs.libframework }} --configuration=Release ${{ matrix.component }}
+      - name: Build tests
+        run: dotnet build --no-dependencies --framework ${{ steps.dotnet.outputs.framework }} --configuration=Release  ${{ matrix.component }}.Test
       - name: Run tests
         shell: bash
         run: |
-          cd Consul.Test
-          ./consul agent -dev -config-file test_config.json -log-file consul.log >consul-stdout.log 2>consul-stderr.log &
-          dotnet test --configuration=Release --logger "GitHubActions;report-warnings=false" --no-build -v=Normal
+          cd ${{ matrix.component }}.Test
+          if [ "${{ matrix.component }}" == "Consul" ]
+          then
+            ./consul agent -dev -config-file test_config.json -log-file consul.log >consul-stdout.log 2>consul-stderr.log &
+          fi
+          if [ "${{ steps.dotnet.outputs.framework }}" != "netcoreapp2.1" ]
+          then
+            logger='--logger "GitHubActions;report-warnings=false"'
+          fi
+          dotnet test --framework ${{ steps.dotnet.outputs.framework }} --configuration=Release --logger "GitHubActions;report-warnings=false" --no-build -v=Normal
       - name: Upload Consul logs
-        if: failure()
+        if: failure() && matrix.component == 'Consul'
         uses: actions/upload-artifact@v2
         with:
-          name: consul-logs-${{ matrix.os }}
+          name: consul-logs-${{ matrix.os }}_dotnet-${{ steps.dotnet.outputs.version }}_consul-${{ matrix.consul }}
```

Comment:
Why we use `dotnet.outputs.version` here, isn't `dotnet.outputs.framework` a better choice ?